### PR TITLE
Set product image if combination image is null in displayAjaxProductsList 

### LINKED ISSUE #20001: Product pack missing images on combination product
First of all sorry for my bad English. 

#### Describe the bug

When I make a product pack with a combination product the image is missing  if I not select the image for these combination. (URL missing ID I think)
https://domain.com/-home_default/product.jpg 404

#### Expected behavior

If I not select any image for a combination product (ex. if the product image it's the same for all combinations) I expect that the default image of the product be shown by default.

**Screenshots**

Product combination selected: 
image

Creating Product Pack: 
image

#### Additional information

* PrestaShop version: 1.7.6.4
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by JBassx**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | When create a product pack, If any image is selected for a combination product (ex. if the product image it's the same for all combinations) the default image of the product will be shown by default.
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #20001 .
| How to test?  | Create a Product Pack -> Add a product -> You can see now the default image is loaded

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20225)

### CODE REVIEW

File: controllers/admin/AdminProductsController.php [hunk 1]
```diff
@@ -3377,7 +3377,9 @@ public function displayAjaxProductsList()
                                 $results[$combination['id_product_attribute']]['ref'] = !empty($item['reference']) ? $item['reference'] : '';
                             }
                             if (empty($results[$combination['id_product_attribute']]['image'])) {
-                                $results[$combination['id_product_attribute']]['image'] = str_replace('http://', Tools::getShopProtocol(), $context->link->getImageLink($item['link_rewrite'], $combination['id_image'], 'home_default'));
+                                $image_id = $combination['id_image'] ? $combination['id_image'] : $item['id_image'];
```
Comment (Progi1984) [inline controllers/admin/AdminProductsController.php]:
```suggestion
                                $image_id = $combination['id_image'] ?? $item['id_image'];
```

**Review by JBassx — COMMENTED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20225)* status: all files reviewed, 1 unresolved discussion (waiting on @JBassx)

**Review by JBassx — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20225)* status: all files reviewed, 1 unresolved discussion (waiting on @Progi1984)

___
*[controllers/admin/AdminProductsController.php, line 3380 at r1](https://reviewable.io/reviews/prestashop/prestashop/20225#-MCXs4gK4AO4DCMdyIb_-r1-3380:-MDJLj4Z1GYcYqud19lT:b-896fix) ([raw file](https://github.com/prestashop/prestashop/blob/6596e3f624a7e2c54e553bf1fd4428c1ca550cef/controllers/admin/AdminProductsController.php#L3380)):*
<details><summary><i>Previously, Progi1984 (Progi1984) wrote…</i></summary><blockquote>

```suggestion
                                $image_id = $combination['id_image'] ?? $item['id_image'];
```
</blockquote></details>

Done.

**Review by Progi1984 — DISMISSED** [review]

**Review by JBassx — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20225#-:-MDJjV1C0Bk-PbVBw-pB:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @Progi1984)

File: controllers/admin/AdminProductsController.php [hunk 2]
```diff
@@ -3377,7 +3377,9 @@ public function displayAjaxProductsList()
                                 $results[$combination['id_product_attribute']]['ref'] = !empty($item['reference']) ? $item['reference'] : '';
                             }
                             if (empty($results[$combination['id_product_attribute']]['image'])) {
-                                $results[$combination['id_product_attribute']]['image'] = str_replace('http://', Tools::getShopProtocol(), $context->link->getImageLink($item['link_rewrite'], $combination['id_image'], 'home_default'));
+                                $image_id = $combination['id_image'] ?? $item['id_image'];
+                                $results[$combination['id_product_attribute']]['image'] = str_replace('http://', Tools::getShopProtocol(), $context->link->getImageLink($item['link_rewrite'], $image_id, 'home_default'));
+
                             }
```
Comment (atomiix) [inline controllers/admin/AdminProductsController.php]:
```suggestion
                                $results[$combination['id_product_attribute']]['image'] = str_replace('http://', Tools::getShopProtocol(), $context->link->getImageLink($item['link_rewrite'], $image_id, 'home_default'));
                            }
```

**Review by PierreRambaud — APPROVED** [review]

**Review by JBassx — COMMENTED** [review]
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20225)* status: all files reviewed, 2 unresolved discussions (waiting on @JBassx and @Progi1984)

**Review by Progi1984 — APPROVED** [review]

**Review by JBassx — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20225)* status: all files reviewed, 2 unresolved discussions (waiting on @atomiix and @Progi1984)

___
*[controllers/admin/AdminProductsController.php, line 3383 at r2](https://reviewable.io/reviews/prestashop/prestashop/20225#-MCXs4gK4AO4DCMdyIb_-r2-3383:-MDeQoUJ6nJ__CHfcqvx:b-896fix) ([raw file](https://github.com/prestashop/prestashop/blob/bc28a7dd5fbcdec90adc7e56a8d21067591c1fd0/controllers/admin/AdminProductsController.php#L3383)):*
<details><summary><i>Previously, atomiix wrote…</i></summary><blockquote>

```suggestion
                                $results[$combination['id_product_attribute']]['image'] = str_replace('http://', Tools::getShopProtocol(), $context->link->getImageLink($item['link_rewrite'], $image_id, 'home_default'));
                            }
```
</blockquote></details>

Done.

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @JBassx,

I tried with this PR (fresh install) but nothing changed.
Same error exists
image
image

Thanks to check and feedback.

**Comment by PierreRambaud** [comment]
Hi @JBassx ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**